.github/workflows/ci.yml

```
@@ -150,6 +152,24 @@ jobs:
         rustup target add ${{ matrix.target }}
         cargo build --verbose --target ${{ matrix.target }}
 
+  test-freebsd:
+    runs-on: macos-12
```

Could 
```suggestion
    runs-on: ubuntu-latest
```

Seems like the Ubuntu jobs are faster than mac (I assume there's a smaller quantity of them).

---

Unfortunately VM Actions only supports macos-12, as nested virtualisation isn't available elsewhere (at least on the free tier).